tslint.json

```diff
@@ -8,7 +8,12 @@
     "interface-name": [true, "never-prefix"],
     "no-shadowed-variable": false,
     "no-string-literal": false,
-    "trailing-comma": false,
-    "member-ordering": ["fields-first"]
+    "trailing-comma": [false],
+    "member-ordering": ["fields-first"],
+    "array-type": ["array-simple"],
+    "arrow-parens": false,
+    "max-classes-per-file": [false],
+    "prefer-for-of": false,
+    "ban-types": false

```

Comment:
This works somewhat. 

`() => void` will not work but `() => any` will.

The exception to this is dynamic-styles.ts

``` typescript
export interface StyleModule {
  [x: string]: () => any;
}

export function createDynamicStyleModule(styleUrl: string): StyleModule {
  @styles()
  @useStyles(styleUrl)
  class DynamicTheme {}

  return {
    [getNextDynamicThemeId()]: DynamicTheme
  };
}
```

Which causes the error.

```
src/styles/dynamic-styles.ts(18,3): error TS2322: Type '{ [x: string]: typeof DynamicTheme; }' is not assignable to type 'StyleModule'.
[4]   Index signatures are incompatible.
[4]     Type 'typeof DynamicTheme' is not assignable to type '() => any'.
[4]       Type 'typeof DynamicTheme' provides no match for the signature '(): any'
```

I am still fairly new to Typescript overall, so I am not entirely sure how to handle that. I have tried to set it as `() => DynamicTheme` but that did not work as expected.